WordPress/openverse
# Update Nuxt i18n

## Fixes



## Description


This PR updates Nuxt i18n, and adds the necessary changes:
- the locale files are now located in `frontend/i18n/locales/` instead of `frontend/src/locales/`
- the object returned from `useI18n` is now typed differently (uses the locales available in the app), and it's difficult to get it correct in all environment (unit/Playwright tests, type checks and production), so I refactored the functions that accept `i18n` object to accept `i18n.t` instead.
- the fallback locales were removed when migrating to Nuxt 3; added back in this PR.
- the custom pluralization rules created in the follow up PR #5177 (from the data downloaded from GlotPress) added.

The unused i18n-check script was removed

## Testing Instructions

The CI should pass, and the app should work correctly without warning.
Run the i18n scripts and check that they all work. Delete locale files between running the scripts.
- `ov just frontend/run i18n:en`
- `ov just frontend/run i18n:copy-test-locales`
- `ov just frontend/run i18n`
- `ov just frontend/run i18n:copy-test-locales`
- `ov just frontend/run i18n:no-get` (without removing the existing locales: check that they are reused and saved as `valid-locales.json`)
- `ov just frontend/run i18n:create-locales-list`
- `ov just frontend/run i18n:get-translations`
- `ov just frontend/run i18n:update-locales`
- `ov just frontend/run i18n:generate-pot`


## Checklist


      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
